### PR TITLE
3.3: Improve and extend the security checklist

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | 3.3
| Edition       | All

#2565 for 3.3

1fd64577aa1fa00a2431304761774983cf0f4a29 fully overrides the previous version.

Let's check if there is something not backward compatible.

Preview: https://ez-systems-developer-documentation--2571.com.readthedocs.build/en/2571/guide/security_checklist/

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/guide/security_checklist.md [hunk 1]
```diff
@@ -4,65 +4,192 @@ description: Ensure that your Ibexa DXP installation is secure by following our
...
+- By default, a [Powered-By header](https://doc.ibexa.co/en/latest/update_and_migration/from_1.x_2.x/update_db_to_2.5/#powered-by-header) is set. It specifies what version of the DXP is running. For example, `x-powered-by: [[= product_name_exp =]] v4`. This doesn't expose anything that couldn't be detected through other means. But if you wish to obscure this, you can either omit the version number, or disable the header entirely by setting `enabled: false`.
+
+    ```yaml
+    ibexa_system_info:
+      system_info:
+        powered_by:
+          # major => v4 || minor => v4.6 || none
+          release: major
+          # true || false
+          enabled: false
```
Comment (adriendupuis) [inline docs/guide/security_checklist.md:129]: 3.3 might use https://github.com/ezsystems/ez-support-tools/blob/2.3/src/bundle/DependencyInjection/Configuration.php so config name has changed.
I'm investigating closer…

Comment (adriendupuis) [reply docs/guide/security_checklist.md:129]: According to my comparison of https://github.com/ibexa/system-info/blob/v4.6.14/src/bundle/DependencyInjection/Configuration.php and https://github.com/ezsystems/ez-support-tools/blob/v2.3.12/src/bundle/DependencyInjection/Configuration.php only the extension name change.

```suggestion
    ezplatform_support_tools:
      system_info:
        powered_by:
          # major => v4 || minor => v4.6 || none
          release: major
          # true || false
          enabled: false
```

Comment (glye) [reply docs/guide/security_checklist.md:129]: The update doc confirms: https://doc.ibexa.co/en/latest/update_and_migration/from_1.x_2.x/update_db_to_2.5/#d-update-to-v25

File: docs/guide/security_checklist.md [hunk 2]
```diff
@@ -4,65 +4,192 @@ description: Ensure that your Ibexa DXP installation is secure by following our
...
+
+    See [setting up password rules](passwords.md#password-rules).
+
+### Protect against brute force attacks
+
+Consider introducing a measure against brute force login attacks, like CAPTCHA.
+Adjust timeout limits to your needs:
+
+When using the "forgot password" feature, a token is created which expires if the user doesn't click the password reset link that gets mailed to them in time.
+The time before it expires is set in the parameter `ibexa.site_access.config.default.security.token_interval_spec`.
```
Comment (adriendupuis) [inline docs/guide/security_checklist.md:45]: @glye I'm confused with this one.

- https://github.com/ezsystems/ezplatform-user/blob/v2.3.12/src/bundle/Resources/config/ezplatform_default_settings.yaml#L3 looks like the one I'm looking for.
- https://github.com/ezsystems/ezplatform-user/blob/v2.3.12/src/bundle/DependencyInjection/Configuration/Parser/Security.php#L20-L24 looks outdated
- https://doc.ibexa.co/en/3.3/release_notes/ez_platform_v3.0_deprecations/#event-names-changed has it as renamed JS event name?!?

```suggestion
The time before it expires is set in the parameter `ezsettings.default.security.token_interval_spec`.
```

File: docs/guide/security_checklist.md [hunk 3]
```diff
@@ -4,65 +4,192 @@ description: Ensure that your Ibexa DXP installation is secure by following our
...
+as an employee, or the Corporate Portal as a business partner.
+You can send invitations to individual users or in bulk.
+These invitations time out according to the parameter
+`ibexa.site_access.config.default.user_invitation.hash_expiration_time`.
+This can safely be longer than the "forgot password" time, since attackers cannot generate invitations.
+Don't leave it longer than it needs to be, though.
+
+These timeouts are both entered as [PHP DateInterval duration strings](https://www.php.net/manual/en/dateinterval.construct.php).
+The forgot password feature defaults to "PT1H" (one hour).
+The account invitation feature defaults to "P7D" (seven days).
```
Comment (adriendupuis) [inline docs/guide/security_checklist.md:62]: This feature seems added circa 4.2 according to
https://github.com/ibexa/user/pull/35/files#diff-d7dd8ed9e114fd1e91bb9a9c4ee717a918265fa26cd2d152dbd14a909408fd59R40 and its ticket.

The remaining part may need rewording.

```suggestion
This timeout is entered as [PHP DateInterval duration strings](https://www.php.net/manual/en/dateinterval.construct.php).
The forgot password feature defaults to "PT1H" (one hour).
```

File: docs/guide/security_checklist.md [hunk 4]
```diff
@@ -4,65 +4,192 @@ description: Ensure that your Ibexa DXP installation is secure by following our
...
+
+### Disable Varnish when using Fastly
+
+If you're using Fastly, disable Varnish.
+See [Security advisory: EZSA-2020-002](https://developers.ibexa.co/security-advisories/ezsa-2020-002-unauthorised-cache-purge-with-misconfigured-fastly).
+
+### Block upload of unwanted file types
+
+The `ibexa.site_access.config.default.io.file_storage.file_type_blacklist` setting is defined in the config file `src/bundle/Core/Resources/config/default_settings.yml` in the Core bundle.
```
Comment (adriendupuis) [inline docs/guide/security_checklist.md:71]:
```suggestion
The `ezsettings.default.io.file_storage.file_type_blacklist` setting is defined in the config file `src/bundle/Core/Resources/config/default_settings.yml` in the Core bundle.
```

File: docs/guide/security_checklist.md (hunk 3 above)
Comment (glye) [reply docs/guide/security_checklist.md:62]: Nice catch. Edit looks good. Maybe "as a [PHP DateInterval duration string]"

File: docs/guide/security_checklist.md [hunk 5]
```diff
@@ -4,65 +4,183 @@ description: Ensure that your Ibexa DXP installation is secure by following our
...
+
+When using the "forgot password" feature, a token is created which expires if the user doesn't click the password reset link that gets mailed to them in time.
+The time before it expires is set in the parameter `ezsettings.default.security.token_interval_spec`.
+By nature this feature must be available to users before they have logged in, including would-be attackers.
+If an attacker uses this feature with someone else's email address, the attacker doesn't receive the email.
+But they could still try to guess the password reset link.
+That's why this interval should be as short as possible.
+5 minutes is often enough.
+
+This timeout is entered as a [PHP DateInterval duration strings](https://www.php.net/manual/en/dateinterval.construct.php).
```
Comment (glye) [inline docs/guide/security_checklist.md:52]:
```suggestion
This timeout is entered as a [PHP DateInterval duration string](https://www.php.net/manual/en/dateinterval.construct.php).
```

**Review by glye — APPROVED** [review]
+1 with the "strings" change. Thanks!

**Pull request merged**